### PR TITLE
Delete residual file caused by network interruption and so on. Fixes …

### PR DESCRIPTION
**Pull request by Lighting-wings**
…#3204

Signed-off-by: huangkai <huangkai76@huawei.com>

### CODE REVIEW

**Review by dsu-igeek — APPROVED** [review]
LGTM

**Review by jenting — CHANGES_REQUESTED** [review]
Please add a changelog for this PR, thank you.

File: pkg/controller/restore_controller.go [hunk 1]
```diff
@@ -552,6 +552,9 @@ func downloadToTempFile(backupName string, backupStore persistence.BackupStore,
```
[the remark this answers is not in the document]
Comment (nrb) [reply pkg/controller/restore_controller.go:552]: I don't think so - that deletes the file before the `downloadToTempFile` function returns, even if there were no errors, so callers would never get an actual file.

**Review by nrb — DISMISSED** [review]
@Lighting-wings The code looks correct to me, I think the only thing that's missing is a changelog file.

See https://velero.io/docs/v1.5/code-standards/#adding-a-changelog for how we use them.

File: pkg/controller/restore_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenting) [reply pkg/controller/restore_controller.go:552]: You're right, thank you for the explanation.

**Comment by jenting** [comment]
Ping @Lighting-wings

**Comment by dsu-igeek** [comment]
@Lighting-wings Please add a changelog and we will merge this.  Thanks!

**Review by zubron — APPROVED** [review]

**Pull request merged**